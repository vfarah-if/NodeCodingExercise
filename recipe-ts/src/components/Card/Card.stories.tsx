import React from 'react';
import { Story, Meta } from '@storybook/react/types-6-0';
import Card, { CardProps } from './Card';

export default {
	title: 'Component/Card',
	component: Card,
} as Meta;

export const DefaultCard: Story<CardProps> = () => (
	<>
		<Card>
			<div className="container">
				<h4>
					<b>John Doe</b>
				</h4>
				<p>Architect & Engineer</p>
				<img
					src="https://www.w3schools.com/howto/img_avatar.png"
					alt="Avatar"
				/>
			</div>
		</Card>
	</>
);
