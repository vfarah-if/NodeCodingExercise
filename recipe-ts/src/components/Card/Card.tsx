
import React from 'react';
import './style/index.css';

export interface CardProps {
    children?: React.ReactNode;
}

const Card: React.FC<CardProps> = (props) => {
    return <li className="card">{props.children}</li>;
}

export default Card;
