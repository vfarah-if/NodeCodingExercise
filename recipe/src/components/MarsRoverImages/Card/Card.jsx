
import React from 'react';
import './style/index.css';

const Card = (props) => {
    return <li className="card">{props.children}</li>;
};

export default Card
