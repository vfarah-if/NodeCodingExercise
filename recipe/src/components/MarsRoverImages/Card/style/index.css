.card {
	position: relative;
	box-shadow: 0 3px 6px rgba(0, 0, 0, 0.5);
}

.card:hover {
	cursor: pointer;
}

.card img:not(:first-of-type) {
	position: absolute;
	top: 0;
	right: 0;
	bottom: 0;
	left: 0;
	opacity: 0;
}

.card img.is-visible {
	opacity: 1;
}

.container {
	padding: 2px 16px;
}

li {
	list-style: none;
}
