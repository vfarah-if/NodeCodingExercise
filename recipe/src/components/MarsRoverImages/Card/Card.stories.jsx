import React from 'react';
import Card from './Card';

// eslint-disable-next-line import/no-anonymous-default-export
export default {
	title: 'Component/Card',
	component: Card,
};

export const CardStory = () => (
	<>
		<Card>
			<div className="container">
				<h4>
					<b>John Doe</b>
				</h4>
				<p>Architect & Engineer</p>
				<img
					src="https://www.w3schools.com/howto/img_avatar.png"
					alt="Avatar"
				/>
			</div>
		</Card>
	</>
);
