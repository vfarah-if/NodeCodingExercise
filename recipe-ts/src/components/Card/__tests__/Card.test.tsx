import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { DefaultCard } from '../Card.stories';

describe('Card', () => {
	it('should create a card with children contents', () => {
		const { container } = render(<DefaultCard />);

		expect(container).toMatchSnapshot();
	});
});
