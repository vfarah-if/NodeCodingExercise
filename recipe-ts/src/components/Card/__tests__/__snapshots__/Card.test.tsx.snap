// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should create a card with children contents 1`] = `
<div>
  <li
    class="card"
  >
    <div
      class="container"
    >
      <h4>
        <b>
          John Doe
        </b>
      </h4>
      <p>
        Architect & Engineer
      </p>
      <img
        alt="Avatar"
        src="https://www.w3schools.com/howto/img_avatar.png"
      />
    </div>
  </li>
</div>
`;
